src/backend/mod.rs

```diff
@@ -763,6 +788,9 @@ pub fn compile(module: &Module<'_>) -> anyhow::Result<Vec<u8>> {
     }
     names.functions(&func_names);
     into_mod.section(&names);
+    for (k, v) in module.custom_sections.iter() {
+        into_mod.section(&CustomSection { name: &k, data: &v });
```

Comment:
I definitely understand the usefulness of preserving custom sections, but my main concern here is that some custom sections will refer to code or other contents of the module that we will have modified (or completely regenerated), and will no longer be valid. For example, a debug (DWARF info) section will no longer have valid offsets, and will at least confuse, if not crash, a debugger that tries to use it in conjunction with our output.

Is there a particular kind of custom section you need to preserve? Perhaps we could have an allow-list of custom sections that remain valid and thus are preserved?